File: app/src/main/java/org/worshipsongs/domain/DragDrop.java
package org.worshipsongs.domain;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import org.apache.commons.lang3.StringUtils;
import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

import java.util.ArrayList;

/**
 * Author : Madasamy
 * Version : 4.x.x
 */

public class DragDrop
{
    private long id;
    private String title;
    private boolean checked;

    public DragDrop(long id, String title, boolean checked)
    {
        this.id = id;
        this.title = title;
        this.checked = checked;
    }

    public long getId()
    {
        return id;
    }

    public void setId(long id)
    {
        this.id = id;
    }

    public String getTitle()
    {
        return title;
    }

    public void setTitle(String title)
    {
        this.title = title;
    }

    public boolean isChecked()
    {
        return checked;
    }

    public void setChecked(boolean checked)
    {
        this.checked = checked;
    }

    @Override
    public String toString()
    {
        return "DragDrop{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", checked=" + checked +
                '}';
    }

    @Override
    public boolean equals(Object object)
    {
        if (object instanceof DragDrop) {
            DragDrop otherObject = (DragDrop) object;
            EqualsBuilder builder = new EqualsBuilder();
            builder.append(getTitle(), otherObject.getTitle());
            return builder.isEquals();
        }
        return false;
    }

    @Override
    public int hashCode()
    {
        HashCodeBuilder hashCodeBuilder = new HashCodeBuilder();
        hashCodeBuilder.append(getTitle());
        return hashCodeBuilder.hashCode();
    }

    public static String toJson(ArrayList<DragDrop> items)
    {
        Gson gson = new Gson();
        return gson.toJson(items);
    }

    public static ArrayList<DragDrop> toArrays(String jsonString)
    {
        if (StringUtils.isNotBlank(jsonString)) {
            Gson gson = new Gson();
            java.lang.reflect.Type type = new TypeToken<ArrayList<DragDrop>>()
            {
            }.getType();
            return gson.fromJson(jsonString, type);
        }
        return new ArrayList<>();
    }

}

File: app/src/main/java/org/worshipsongs/domain/DragDrop.kt
package org.worshipsongs.domain

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

import org.apache.commons.lang3.StringUtils
import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder

import java.util.ArrayList

/**
 * Author : Madasamy
 * Version : 3.x.x
 */

open class DragDrop
{
    //TODO: Rename this as orderId (Ref: Favourite.kt). (Please note that this will have an impact in existing favourites)
    var id: Long = 0
    var title: String? = null
    var isChecked: Boolean = false

    constructor()
    {
    }

    constructor(orderId: Long, title: String, checked: Boolean)
    {
        this.id = orderId
        this.title = title
        this.isChecked = checked
    }

    override fun toString(): String
    {
        return "DragDrop{" + "orderId=" + id + ", title='" + title + '\''.toString() + ", checked=" + isChecked + '}'.toString()
    }

    override fun equals(`object`: Any?): Boolean
    {
        if (`object` is DragDrop)
        {
            val otherObject = `object` as DragDrop?
            val builder = EqualsBuilder()
            builder.append(title, otherObject!!.title)
            return builder.isEquals
        }
        return false
    }

    override fun hashCode(): Int
    {
        val hashCodeBuilder = HashCodeBuilder()
        hashCodeBuilder.append(title)
        return hashCodeBuilder.hashCode()
    }

    companion object
    {

        fun toJson(items: List<DragDrop>): String
        {
            val gson = Gson()
            return gson.toJson(items)
        }

        fun toArrays(jsonString: String): ArrayList<DragDrop>
        {
            if (StringUtils.isNotBlank(jsonString))
            {
                val gson = Gson()
                val type = object : TypeToken<ArrayList<DragDrop>>()
                {

                }.type
                return gson.fromJson(jsonString, type)
            }
            return ArrayList()
        }
    }

}
